python/lsst/meas/algorithms/stamps.py

```diff
@@ -292,17 +298,21 @@ class StampsBase(abc.ABC, Sequence):
     A butler can be used to read only a part of the stamps,
     specified by a bbox:
 
-    >>> starSubregions = butler.get("brightStarStamps", dataId, parameters={'bbox': bbox})
+    >>> starSubregions = butler.get(
+            "brightStarStamps",
+            dataId,
+            parameters={"bbox": bbox}
+        )
     """
 
-    def __init__(self, stamps, metadata=None, use_mask=True, use_variance=True,
-                 use_archive=False):
-        if not hasattr(stamps, '__iter__'):
-            raise ValueError('The stamps parameter must be iterable.')
+    def __init__(self, stamps, metadata=None, use_mask=True, use_variance=True, use_archive=False):
+        if not hasattr(stamps, "__iter__"):
+            raise ValueError("The stamps parameter must be iterable.")
         for stamp in stamps:
             if not isinstance(stamp, AbstractStamp):
-                raise ValueError('The entries in stamps must inherit from AbstractStamp.  '
-                                 f'Got {type(stamp)}.')
+                raise ValueError(
+                    "The entries in stamps must inherit from AbstractStamp.  " f"Got {type(stamp)}."
+                )
```

Comment:
Similarly, this check may also not be required and is inconsistent with duck typing. But if you choose to keep it, then the second string needs to be merged with the first.

Reply:
Modulo fixing the black string formatting problem here, I don't understand what the problem is with this test? We are requiring that everything is an AbstractStamp as a way to fail early. It's effectively promising the interface.

Reply:
I don't think it's Pythonic to do such `isinstance` checks, and we don't check that the object passed is of the correct type in most of our code. It makes sense only if there are other classes that will conform to this interface, yet will do something completely different. This is likely the case where the original author saw a different codebase and mimicked the same structure here. Is failing early here (occassionally) advantageous over checking this every time at run time?

Reply:
I think it may have been my idea to add this check to be defensive. We were getting programming bugs where people were handing the wrong thing to the constructor and it was hard to track down where the bad stamp was coming from.

Reply:
All the more reason to start using static type checking more broadly :-)

If the branching had to depend on the type of the object, then an `isinstance` check makes sense. If it is simply to check if it is the right type, then these should be left to mypy. What's preventing us from enabling mypy in all our repos, with one function with type hints just to get the ball rolling?

Reply:
It is unlikely that the entire science pipelines is going to be mypy compliant any time soon and it's not clear to me that the science pipelines group have decided they want mypy.

I don't understand why you don't want to check that these are all subclasses of Stamp.

Reply:
I'm reluctant to remove this `isinstance` check without a broader understanding of how this may be being used elsewhere in the stack. As it is, I like the idea of failing fast if the wrong type is being passed in, so I lean towards maintaining this behavior for the time being. Happy to be persuaded it's safe to remove this however.

Reply:
I want it to stay. I added this with Simon because we reasoned that it was better to complain early rather than late and we really want Stamps here.

Reply:
2 against 1 here. May democracy prevail.